mcdir/main.go

```diff
@@ -130,6 +156,16 @@ func (dir *Directory) lookupPeer(pid p2p_peer.ID) (p2p_pstore.PeerInfo, bool) {
 	return pinfo, ok
 }
 
+func (dir *Directory) listPeers() []string {
+	dir.mx.Lock()
```

Comment:
for the mutex? You have to, maps in go are not thread-safe by concious (and correct imo) design decision.